arches_querysets/datatypes/file.py

```diff
@@ -0,0 +1,70 @@
+import copy
+
+from django.utils.translation import get_language
+
+from arches.app.datatypes import datatypes
+from arches.app.utils.i18n import rank_label
+
+
+class FileListDataType(datatypes.FileListDataType):
+    def transform_value_for_tile(self, value, *, languages, **kwargs):
+        if not value:
+            return value
+
+        stringified_list = ",".join([file_info.get("name") for file_info in value])
+        final_value = self.transform_value_for_tile(
+            stringified_list, languages=languages, **kwargs
+        )
+
+        for file_info in final_value:
+            for key, val in file_info.items():
+                if key not in {"altText", "attribution", "description", "title"}:
+                    continue
+                original_val = val
+                if not isinstance(original_val, dict):
+                    file_info[key] = {}
+                for lang in languages:
+                    if lang.code not in file_info[key]:
+                        file_info[key][lang.code] = {
+                            "value": (
+                                original_val if lang.code == get_language() else ""
+                            ),
+                            "direction": lang.default_direction,
+                        }
+
+        return final_value
+
+    def merge_tile_value(self, tile, node_id_str, transformed) -> None:
```

Comment:
why have is separate from `transform_value_for_tile`?

Reply:
Good Q. I was thinking to keep the responsibilities as small as possible. `transform...` is about parsing the input data, and exists on every datatype. `merge...` is about merging incoming single localized language values back into the full set, and is a concept that only exists in arches-querysets for now.

As a project implementer who might let's say override `transform...()` in order to accept legacy id strings, I'd rather not have the burden of reimplementing all the merge logic, too.

This is preliminarily documented here, but I'll add docstrings inline with merge_tile_value.

https://github.com/archesproject/arches-querysets/blob/893c8845c1549e330a327c3f4d388b6180e5cb30/arches_querysets/bulk_operations/tiles.py#L301-L313

Reply:
(Next up this week should be a PR to add some view tests to show how the backend processes these different incoming requests.)